pkg/netdevice/netDeviceProvider.go

```diff
@@ -174,9 +175,12 @@ func (np *netDeviceProvider) GetFilteredDevices(devices []types.PciDevice, rc *t
 
 	// filter for rdma devices
 	if nf.IsRdma {
+		if _, err := os.Stat("/dev/infiniband"); err != nil {
```

Comment:
I think the netDeviceProvider should not be aware of RDMA spec internal details. (i,e the knowledge of `/dev/infiniband`)

i think that a better place to log the absence of the RDMA char devsis in `rdma.go:NewRdmaSpec()` where if we get no char devices we log a warning.

also perhaps in the log add some hint towards what might cause it and solution

Reply:
moved to rdma.go

Reply:
@adrianchiris If moving `/dev/infiniband` check in rdma.go, then evey device in a non-rdma environment will hit this warning message, which might be confusing even if rdma is not configured. Since `NewRdmaSpec` is called in device discover phase (where resourceConfig is not passed in), we cannot detect whether `isRdma` is configured or not by user, so it's not easy to add condition for warning message.

I suggested to add it in `GetFilteredDevices` because we already have a condition there for `isRdma`.

Reply:
The code proposes to perform the check at the moment the RDMA Spec is requested.
without pushing this logic to upper layers.

Its true it has the drawback of reporting on every call. (which i originally weighted against having the logic in GetFilteredDevices)

its possible to run the check once or move the logic back to upper layer if thats acceptable.

Reply:
When looking at this [proposal](https://github.com/intel/sriov-network-device-plugin/pull/213), it seems to me that rdma spec could also be retrieved when generating resource pool (similar to getting device spec), where a conditional check on `isRdma` can be applied.

Reply:
lets move this to its original location where you proposed.
if/once pool specific logic is introduced, this can then be moved there.